packages/flutter/lib/src/material/material_state.dart

```diff
@@ -66,6 +68,163 @@ enum MaterialState {
 /// set of states.
 typedef MaterialPropertyResolver<T> = T Function(Set<MaterialState> states);
 
+/// A widget that uses [builder] to obtain its child widget.
```

Comment:
I think the first sentence should be more specific and talk about the actual point of this widget.

Reply:
Sure, how does this sound?
```
/// A widget that can be used to create widgets with custom interactive behavior,
/// such as a button that changes its label text style on hover/focus/press.
```